From 51f3c26190f91981e35671ac87bb716dac8a9b7f Mon Sep 17 00:00:00 2001
From: Vernon Mauery <vernon.mauery@intel.com>
Date: Wed, 27 Feb 2019 15:22:52 -0800
Subject: [PATCH] create_pen_list: only print if values are set

On a failed download of the PEN list, the create_pen_list script
improperly printed an invalid entry of { , "" } causing the build to
fail. The last line print must check that it has something to print or
it will print the wrong thing.

Partially resolves ipmitool/ipmitool#11

Signed-off-by: Vernon Mauery <vernon.mauery@intel.com>
---
 lib/create_pen_list | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/create_pen_list b/lib/create_pen_list
index e2b44d2..81864ad 100755
--- a/lib/create_pen_list
+++ b/lib/create_pen_list
@@ -67,7 +67,9 @@ parse_pen_list() {
       }
 
       END {
-          print "{ " PEN ", \"" ENTERPRISE "\" },"
+          if(PEN) {
+            print "{ " PEN ", \"" ENTERPRISE "\" },"
+          }
       }'
 }
 
-- 
2.17.1

